packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -285,6 +298,19 @@ export function executeStepDependencyGraph<
       }
 
       await context.jobState.flush();
+
+      if (context.jobState.waitUntilUploadsComplete) {
+        try {
+          // Failing to upload all integration data should not be considered a
+          // fatal failure. We just want to make this step as a partial success
+          // and move on with our lives!
+          await context.jobState.waitUntilUploadsComplete();
+        } catch (err) {
+          context.logger.stepFailure(step, err);
+          status = StepResultStatus.FAILURE;
```

Comment:
Kind of strange that there is no differentiation between a failure on the step and a failure when attempting to persist the data. Idk if it is worth adding a way to differentiate that or not.